ruby_event_store/lib/ruby_event_store/correlated_events.rb

```diff
@@ -0,0 +1,10 @@
+module RubyEventStore
+  module CorrelatedEvents
+    def self.metadata_for(event)
+      {
+        correlation_id: event.metadata[:correlation_id] || event.event_id,
```

Comment:
@paneq this logic is taken from the existing code-base. However, the `Event#correlate_with` method refers to `event.correlation_id` instead of accessing `metadata`. Could/should this use the method instead of hash key, and maybe we also use this helper within `Event#correlate_with`? (or in some form share this logic there as well)

Reply:
@paneq I ended up renaming this to `CorrelatedMessages` and implemented an approach to my previous comment. Let me know your thoughts.

Thanks!